<h1>ACTIVITY: PALMER PENGUINS DATA ANALYTICS</h1> 
<hr>
<h2>Analyst: John Allen Bonggay</h2>

In [13]:
#libraries

In [14]:
import numpy as np
import pandas as pd
from scipy.stats import f_oneway
import warnings
warnings.filterwarnings('ignore')

In [15]:
birds = pd.read_csv('Datasets\\penguins.csv')

In [16]:
birds

species     island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
0       Adelie  Torgersen            39.1           18.7              181.0   
1       Adelie  Torgersen            39.5           17.4              186.0   
2       Adelie  Torgersen            40.3           18.0              195.0   
3       Adelie  Torgersen             NaN            NaN                NaN   
4       Adelie  Torgersen            36.7           19.3              193.0   
..         ...        ...             ...            ...                ...   
339  Chinstrap      Dream            55.8           19.8              207.0   
340  Chinstrap      Dream            43.5           18.1              202.0   
341  Chinstrap      Dream            49.6           18.2              193.0   
342  Chinstrap      Dream            50.8           19.0              210.0   
343  Chinstrap      Dream            50.2           18.7              198.0   

     body_mass_g     sex  year  
0         3750.0    male  2007  
1         3800.0  female  2007  
2         3250.0  female  2007  
3            NaN     NaN  2007  
4         3450.0  female  2007  
..           ...     ...   ...  
339       4000.0    male  2009  
340       3400.0  female  2009  
341       3775.0    male  2009  
342       4100.0    male  2009  
343       3775.0  female  2009  

[344 rows x 8 columns]

<h1>Descriptive Analytics</h1>

In [17]:
birds.describe()

bill_length_mm  bill_depth_mm  flipper_length_mm  body_mass_g  \
count      342.000000     342.000000         342.000000   342.000000   
mean        43.921930      17.151170         200.915205  4201.754386   
std          5.459584       1.974793          14.061714   801.954536   
min         32.100000      13.100000         172.000000  2700.000000   
25%         39.225000      15.600000         190.000000  3550.000000   
50%         44.450000      17.300000         197.000000  4050.000000   
75%         48.500000      18.700000         213.000000  4750.000000   
max         59.600000      21.500000         231.000000  6300.000000   

              year  
count   344.000000  
mean   2008.029070  
std       0.818356  
min    2007.000000  
25%    2007.000000  
50%    2008.000000  
75%    2009.000000  
max    2009.000000

In [18]:
birds.describe().T

count         mean         std     min       25%      50%  \
bill_length_mm     342.0    43.921930    5.459584    32.1    39.225    44.45   
bill_depth_mm      342.0    17.151170    1.974793    13.1    15.600    17.30   
flipper_length_mm  342.0   200.915205   14.061714   172.0   190.000   197.00   
body_mass_g        342.0  4201.754386  801.954536  2700.0  3550.000  4050.00   
year               344.0  2008.029070    0.818356  2007.0  2007.000  2008.00   

                      75%     max  
bill_length_mm       48.5    59.6  
bill_depth_mm        18.7    21.5  
flipper_length_mm   213.0   231.0  
body_mass_g        4750.0  6300.0  
year               2009.0  2009.0

<h1>Filtering Data</h1>

In [19]:
birds['island'] == 'Torgersen'

0       True
1       True
2       True
3       True
4       True
       ...  
339    False
340    False
341    False
342    False
343    False
Name: island, Length: 344, dtype: bool

In [20]:
Torgersen_island = birds[birds['island'] == 'Torgersen']

In [21]:
birds['island'].value_counts()

island
Biscoe       168
Dream        124
Torgersen     52
Name: count, dtype: int64

In [22]:
birds['island'] == 'Torgersen'

0       True
1       True
2       True
3       True
4       True
       ...  
339    False
340    False
341    False
342    False
343    False
Name: island, Length: 344, dtype: bool

In [23]:
Torgersen_island['body_mass_g'].mean()

np.float64(3706.372549019608)

<h1>Adding Two Datas</h1>

In [24]:
birds[birds['sex']<= male & birds[birds['island'] == 'Torgersen']

_IncompleteInputError: incomplete input (4081851517.py, line 1)

<h1>Cleaning Data Rows</h1>

In [25]:
birds.isnull()

species  island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
0      False   False           False          False              False   
1      False   False           False          False              False   
2      False   False           False          False              False   
3      False   False            True           True               True   
4      False   False           False          False              False   
..       ...     ...             ...            ...                ...   
339    False   False           False          False              False   
340    False   False           False          False              False   
341    False   False           False          False              False   
342    False   False           False          False              False   
343    False   False           False          False              False   

     body_mass_g    sex   year  
0          False  False  False  
1          False  False  False  
2          False  False  False  
3           True   True  False  
4          False  False  False  
..           ...    ...    ...  
339        False  False  False  
340        False  False  False  
341        False  False  False  
342        False  False  False  
343        False  False  False  

[344 rows x 8 columns]

In [26]:
birds.isnull().sum()

species               0
island                0
bill_length_mm        2
bill_depth_mm         2
flipper_length_mm     2
body_mass_g           2
sex                  11
year                  0
dtype: int64

<h1>Filling Up Null Values</h1>

In [27]:
birds['bill_length_mm'] = birds['bill_length_mm'].fillna(birds['bill_length_mm'].mean())
birds['bill_depth_mm'] = birds['bill_depth_mm'].fillna(birds['bill_depth_mm'].mean())
birds['flipper_length_mm'] = birds['flipper_length_mm'].fillna(birds['flipper_length_mm'].mean())
birds['body_mass_g'] = birds['body_mass_g'].fillna(birds['body_mass_g'].mean())
birds['sex'] = birds['sex'].fillna(birds['sex'].mode()[0])

<h1>Changing Data Types</h1>

In [28]:
birds.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 344 entries, 0 to 343
Data columns (total 8 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   species            344 non-null    object 
 1   island             344 non-null    object 
 2   bill_length_mm     344 non-null    float64
 3   bill_depth_mm      344 non-null    float64
 4   flipper_length_mm  344 non-null    float64
 5   body_mass_g        344 non-null    float64
 6   sex                344 non-null    object 
 7   year               344 non-null    int64  
dtypes: float64(4), int64(1), object(3)
memory usage: 21.6+ KB


In [29]:
birds['species'] = birds['species'].astype('category')
birds['island'] = birds['island'].astype('category')
birds['sex'] = birds['sex'].astype('category')
birds['year'] = birds['year'].astype('category')

In [30]:
birds.describe()

bill_length_mm  bill_depth_mm  flipper_length_mm  body_mass_g
count      344.000000     344.000000         344.000000   344.000000
mean        43.921930      17.151170         200.915205  4201.754386
std          5.443643       1.969027          14.020657   799.613058
min         32.100000      13.100000         172.000000  2700.000000
25%         39.275000      15.600000         190.000000  3550.000000
50%         44.250000      17.300000         197.000000  4050.000000
75%         48.500000      18.700000         213.000000  4750.000000
max         59.600000      21.500000         231.000000  6300.000000

<h1>DATA ANALYZING</h1>

<h3>1. How Many Male Penguins Are There In Torgensen Island?</h3>

In [31]:
birds['island'].value_counts()

island
Biscoe       168
Dream        124
Torgersen     52
Name: count, dtype: int64

In [32]:
birds[birds['island'] == 'Torgersen']['sex'].value_counts()

sex
male      28
female    24
Name: count, dtype: int64

<h4>Insight 1</h4>
<p>Male penguins that are living in Torgersen Island are in a total of 28.</p>

<h3>2. How Many Female Penguins In Torgersen Island?</h3>

In [33]:
birds[birds['island'] == 'Torgersen']['sex'].value_counts()

sex
male      28
female    24
Name: count, dtype: int64

<h4>Insight 2</h4>
<p>The number of Female penguin residing in Torgersen island sums to 24, which are less than 4 numbers to the male penguins.</p>

<h3>3. How many Penguins Are There In Torgersen Island?</h3>

In [34]:
birds[birds['island'] == 'Torgersen']['species'].value_counts()

species
Adelie       52
Chinstrap     0
Gentoo        0
Name: count, dtype: int64

<h4>Insight 3</h4>
<p>The number of penguin living in Torgersen island sums a total of 52, which all belong to the Adelie Specie.</p>

<h1>4. What Species Are The Most Dominant in Numbers Among All Species?</h1>

In [35]:
birds['species'].value_counts()

species
Adelie       152
Gentoo       124
Chinstrap     68
Name: count, dtype: int64

<h4>Insight 4</h4>
<p>The most dominant specie comes from Adelie summing to 152 penguins, 2nd is from Gentoo(124), and lastly is Chinstrap(68).</p>

<h1>5. What Is The Average Bill Length of Penguins In Biscoe Island?</h1>

In [36]:
birds['bill_length_mm'] = birds['bill_length_mm'].fillna(birds['bill_length_mm'].mean())

In [37]:
bills = birds.groupby(['species', 'island'])['bill_length_mm'].mean()

In [38]:
bills

species    island   
Adelie     Biscoe       38.975000
           Dream        38.501786
           Torgersen    39.046576
Chinstrap  Biscoe             NaN
           Dream        48.833824
           Torgersen          NaN
Gentoo     Biscoe       47.475983
           Dream              NaN
           Torgersen          NaN
Name: bill_length_mm, dtype: float64

<h4>Insight 5</h4>
<p>The average bill length of penguins residing on Biscoe Island is 47.48.</p>

<h1>6.Is there a correlation between bill depth to penguins body mass?</h1>

In [39]:
birds.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 344 entries, 0 to 343
Data columns (total 8 columns):
 #   Column             Non-Null Count  Dtype   
---  ------             --------------  -----   
 0   species            344 non-null    category
 1   island             344 non-null    category
 2   bill_length_mm     344 non-null    float64 
 3   bill_depth_mm      344 non-null    float64 
 4   flipper_length_mm  344 non-null    float64 
 5   body_mass_g        344 non-null    float64 
 6   sex                344 non-null    category
 7   year               344 non-null    category
dtypes: category(4), float64(4)
memory usage: 12.7 KB


In [40]:
corre = birds['bill_depth_mm'].corr(birds['body_mass_g'])

In [41]:
corre

np.float64(-0.4719156211860664)

<h3>Insight 6</h3>
<P>There is no correlation between bill depth to a penguin's body mass.</P>

<h1>7. Is there a correlation between bill length to flipper length?</h1>

In [42]:
bill_l = birds['bill_length_mm'].corr(birds['flipper_length_mm'])

In [43]:
bill_l

np.float64(0.6561813407464278)

<h3>Insight 7</h3>
<p>There is a correlation between the penguins's bill length to it's flipper length, yet the data seems scarce to analyze what is the reason
for it's correlation?</p>

<h1>8. What Species has the heaviest average body mass and from what year?</h1>

In [44]:
heaviest_penguins = birds.loc[birds.groupby(['year', 'species'])['body_mass_g'].idxmax()]
heaviest_penguins[['year', 'species', 'body_mass_g']]

year    species  body_mass_g
7    2007     Adelie       4675.0
295  2007  Chinstrap       4400.0
169  2007     Gentoo       6300.0
81   2008     Adelie       4700.0
313  2008  Chinstrap       4800.0
229  2008     Gentoo       6000.0
109  2009     Adelie       4775.0
321  2009  Chinstrap       4450.0
269  2009     Gentoo       6000.0

<h3>Insight 8</h3>
<P>The heaviest species comes from Gentoo weighing 6000.00g average body mass from year 2008 and 2009.</P>

<h1>9.How many penguins are there for each combination of sex and species??</h1>

In [45]:
birds.groupby(['sex', 'species']).size()

sex     species  
female  Adelie       73
        Chinstrap    34
        Gentoo       58
male    Adelie       79
        Chinstrap    34
        Gentoo       66
dtype: int64

<h3>Insight 9</h3>
<P>The Adelie sums up to 152, Chinstrap(68), and Gentoo(124).</P>

<h1>10.What is the average flipper length for each island and sex??</h1>

In [46]:
birds.groupby(['island', 'sex'])['flipper_length_mm'].mean()

island     sex   
Biscoe     female    205.687500
           male      213.260400
Dream      female    190.016393
           male      196.031746
Torgersen  female    188.291667
           male      194.032686
Name: flipper_length_mm, dtype: float64

<h3>Insight 10</h3>
<P>This shows the difference sizes between different penguin sexes and their responsive islands.</P>

<h1>11.Is there a correlation between male and female bill length?</h1>

In [47]:
birds.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 344 entries, 0 to 343
Data columns (total 8 columns):
 #   Column             Non-Null Count  Dtype   
---  ------             --------------  -----   
 0   species            344 non-null    category
 1   island             344 non-null    category
 2   bill_length_mm     344 non-null    float64 
 3   bill_depth_mm      344 non-null    float64 
 4   flipper_length_mm  344 non-null    float64 
 5   body_mass_g        344 non-null    float64 
 6   sex                344 non-null    category
 7   year               344 non-null    category
dtypes: category(4), float64(4)
memory usage: 12.7 KB


In [48]:
birds['sex'] == 'male'

0       True
1      False
2      False
3       True
4      False
       ...  
339     True
340    False
341     True
342     True
343    False
Name: sex, Length: 344, dtype: bool

In [49]:
birds[birds['sex'] == 'male']['bill_length_mm']

0      39.10000
3      43.92193
5      39.30000
7      39.20000
8      34.10000
         ...   
334    50.20000
336    51.90000
339    55.80000
341    49.60000
342    50.80000
Name: bill_length_mm, Length: 179, dtype: float64

In [50]:
birds[birds['sex'] == 'female']['bill_length_mm']

1      39.5
2      40.3
4      36.7
6      38.9
12     41.1
       ... 
335    45.6
337    46.8
338    45.7
340    43.5
343    50.2
Name: bill_length_mm, Length: 165, dtype: float64

In [51]:
f_oneway(birds[birds['sex'] == 'male']['bill_length_mm'],birds[birds['sex'] == 'female']['bill_length_mm'])

F_onewayResult(statistic=np.float64(39.65455984564769), pvalue=np.float64(9.307501289283404e-10))

<h3>Insight 11</h3>
<P>There is a correlation between penguin's difference in sex and their bill length, must be their functionality as a means for hunting and nurturing offspring.</P>

<h1>12.Which island has the lowest average bill depth of penguins?</h1>

In [53]:
avg = birds.groupby('island')['bill_depth_mm'].mean().sort_values()

In [54]:
avg

island
Biscoe       15.882447
Dream        18.344355
Torgersen    18.404830
Name: bill_depth_mm, dtype: float64

<h3>Insight 12</h3>
<P>The island with the lowest average bill depth comes from Biscoe Island, must have been the situation on their habitat.</P>

<h1>Aggregating Data</h1>

In [ ]:
hadouken = birds.groupby(['species', 'sex', 'island']).size()

In [ ]:
hadouken

<h1>CORRELATIONAL</h1>